x-team/xp-ui
# Update Shortcuts

**Release Type:** *Non-breaking changes*

Fixes https://x-team-internal.atlassian.net/browse/XP-2894

## Description

Upgrade Shortcuts and related components.

## Checklist


## Related PRs

branch | PR
------ | ------
`XP-2894-upgrade-shortcuts-with-updated-genericcopytoclipboard` | [XP](https://github.com/x-team/xp/pull/1999)

## Steps to Test or Reproduce

- http://localhost:9001/?path=/story/ui-components-shortcuts--basic-usage (it has dynamic sample data, so when reloading the page you can check with text of different sizes and observe it doesn't break line and for any case will look strange)
- http://localhost:9001/?path=/story/ui-components-genericcopytoclipboard--with-custom-x-offset (this is not very dynamic solution but works for our case, the small target elements need the Tooltip to be positioned with a bit negative X offset)
- http://localhost:9001/?path=/story/ui-components-svgicon--basic-color (had to update `redirect` and `check` SVGs)

## Impacted Areas in Application

- src/components/ui/GenericCopyToClipboard/GenericCopyToClipboard.stories.js
- src/components/ui/GenericCopyToClipboard/Tooltip.js
- src/components/ui/GenericCopyToClipboard/index.js
- src/components/ui/Shortcuts.js
- src/components/ui/Shortcuts.stories.js
- src/components/ui/SvgIcon.js

## Screenshots

> Something that we seem to skip during the review of https://github.com/x-team/xp-ui/pull/440, the SVG of the `copy` doesn't had the inside coloured correctly:

<img width="49" alt="Screen Shot 2019-09-19 at 09 25 41" src="https://user-images.githubusercontent.com/131859/65255033-8f618b80-dad3-11e9-914e-504c6f25b253.png">

> So I get the SVG from Figma again: /cc @rayepps 

<img width="534" alt="Screen Shot 2019-09-19 at 09 24 44" src="https://user-images.githubusercontent.com/131859/65255032-8ec8f500-dad3-11e9-9ffe-0926d20ee8fd.png">

> Resulted in:

<img width="47" alt="Screen Shot 2019-09-19 at 09 26 05" src="https://user-images.githubusercontent.com/131859/65255035-8f618b80-dad3-11e9-85cd-57ac626500c8.png">

> Also the `redirect` link seemed wrong, we didn't discussed about changing this, but I believe its a good move and worth to check with Paul about it, so far it was used only in 1 place:

<img width="528" alt="Screen Shot 2019-09-19 at 11 31 59" src="https://user-images.githubusercontent.com/131859/65255036-8f618b80-dad3-11e9-85cf-614da0f8bdf9.png">

> But that doesn't match with the expectation for Shortcuts:

<img width="269" alt="Screen Shot 2019-09-19 at 11 32 47" src="https://user-images.githubusercontent.com/131859/65255037-8ffa2200-dad3-11e9-900c-f1717a2f105d.png">

> So this is how it looks with the new SVG for `redirect` in **Storybook**:

![2019-09-19 11 37 46](https://user-images.githubusercontent.com/131859/65255029-8ec8f500-dad3-11e9-93e0-178bca8a79b5.gif)

> And this is how it looks everything in **XP app**:

![2019-09-19 12 10 06](https://user-images.githubusercontent.com/131859/65256806-87efb180-dad6-11e9-8f60-6908bd21abed.gif)
